02-peer-protocol.md

```diff
@@ -787,9 +787,20 @@ is destined, is described in [BOLT #4](04-onion-routing.md).
 #### Requirements
 
 A sending node:
-  - MUST NOT offer `amount_msat` it cannot pay for in the
-remote commitment transaction at the current `feerate_per_kw` (see "Updating
-Fees") while maintaining its channel reserve.
+  - if it is _responsible_ for paying the Bitcoin fee:
+    - MUST NOT offer `amount_msat` if, after adding that HTLC to its commitment
+    transaction, it cannot pay the fee for either the local or remote commitment
+    transaction at the current `feerate_per_kw` while maintaining its channel
+    reserve (see "Updating Fees").
+    - SHOULD NOT offer `amount_msat` if, after adding that HTLC to its commitment
+    transaction, its remaining balance doesn't allow it to pay the fee for a
+    future additional non-dust HTLC at `2*feerate_per_kw` while maintaining its
```

Comment:
Now that we make this `SHOULD`, we can make it clear that a node can make this fee buffer as small/large as it want, no need to choose a constant (although 2 can be recommended)

Reply:
No, If we have no signaling of this value/feature (currently not planned) we should really not encourage to set other values than 2 because we loose calculatability how much we can receive on any channel.

Reply:
Reading this latest iteration of the PR, it seems that B (In the channel A->B) doesn't have to take this constant into account at all:

```
    SHOULD NOT offer `amount_msat` if, once the remote node adds that HTLC to
    its commitment transaction, it cannot pay the fee for the updated local or
    remote transaction at the current `feerate_per_kw` while maintaining its
    channel reserve.
```

which is the same as was previously specced.

Reply:
I agree with @halseth there. I removed the requirement on the fundee, because we can't force fundees to respect this additional reserve (since race conditions exist, there will always be cases where this requirement will not hold). And going in that direction, it makes sense not to hardcode that factor 2 in the spec (otherwise this will simply encourage implementations to *not* be spec-compliant if at some point in the future a factor 2 doesn't seem to be enough).

I understand that from a UX perspective you want to be able to compute how much you can receive, but that calculation will always be flawed. You can't restrict a remote peer's behaviour; your peer could decide to never let the channel dip below 30% on his side, except in some specific cases. Or he could have any other kind of weird behaviour.
That doesn't prevent you from calculating an indication of how much you could receive, if the remote peer behaved in a specific way. That may be slightly different from reality, but it's the best you can get.